### PR TITLE
docs(ts): Fix links

### PR DESCRIPTION
**Pull request by rozzilla**
Fix wrong links/anchors from https://fastify.dev/docs/latest/Reference/TypeScript

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Our link checker is now failing

**Comment by mcollina** [comment]
cc @Fdawgs

**Comment by rozzilla** [comment]
I don't get why the test fails, but currently one link points to `https://fastify.dev/docs/latest/Reference/TypeScript/Validation-and-Serialization.md/` where the correct path is `https://fastify.dev/docs/latest/Reference/Validation-and-Serialization`

**Comment by rozzilla** [comment]
@mcollina I know it's sub-optimal, but to fix the CI check I used an absolute URL now 😉 it seems to work

File: docs/Reference/TypeScript.md [hunk 1]
```diff
@@ -196,7 +196,8 @@ documentation below to learn more about what is available.
 To validate your requests and responses you can use JSON Schema files. If you
 didn't know already, defining schemas for your Fastify routes can increase their
 throughput! Check out the [Validation and
-Serialization](./Validation-and-Serialization.md) documentation for more info.
+Serialization](https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/)
```
Comment (climba03003) [inline docs/Reference/TypeScript.md:199]: You should not use URL here.

File: docs/Reference/TypeScript.md [hunk 2]
```diff
@@ -211,7 +212,7 @@ Fastify offers two packages wrapping `json-schema-to-ts` and `typebox`:
 - `@fastify/type-provider-typebox`
 
 They simplify schema validation setup and you can read more about them in [Type
-Providers](./Type-Providers.md) page.
+Providers](https://fastify.dev/docs/latest/Reference/Type-Providers/) page.
```
Comment (climba03003) [inline docs/Reference/TypeScript.md:215]: You should not use URL here.

**Review by climba03003 — CHANGES_REQUESTED** [review]

**Comment by climba03003** [comment]
If the redirection is wrong, please update the website builder. 
Not the markdown, since it is correct when using `./Validation-and-Serialization.md`

File: docs/Reference/TypeScript.md (hunk 1 above)
Comment (rozzilla) [reply docs/Reference/TypeScript.md]: Done on 24d10bb9 ✅

File: docs/Reference/TypeScript.md (hunk 2 above)
Comment (rozzilla) [reply docs/Reference/TypeScript.md]: Done on 24d10bb9 ✅

**Comment by rozzilla** [comment]
> If the redirection is wrong, please update the website builder. Not the markdown, since it is correct when using `./Validation-and-Serialization.md`

@climba03003 PR updated ✅ 

Can you please point me to the website builder logic, so that I can have a look? 🙏🏼

**Comment by mcollina** [comment]
https://github.com/fastify/website

**Comment by rozzilla** [comment]
I had a look, but I don't find why only the [TypeScript](https://fastify.dev/docs/latest/Reference/TypeScript/) page this issue occurs 🤔 But I confirm that changing the link to `../Validation-and-Serialization` fixes it also when I locally build the `fastify` website, as you can see from the screenshot
image

**Review by climba03003 — APPROVED** [review]

**Review by Uzlopak — APPROVED** [review]
LGTM

**Comment by melroy89** [comment]
> I had a look, but I don't find why only the [TypeScript](https://fastify.dev/docs/latest/Reference/TypeScript/) page has this issue 🤔 Anyway, I confirm that changing the link to `../Validation-and-Serialization` fixes it also when I locally build the `fastify` website, as you can see from the screenshot image
> 
> About the "why" it happens, I don't know, never used `docusaurus` before

Yea indeed, only the TypeScript page has this issue. I don't understand why this page create broken links!??

See issue: https://github.com/fastify/website/issues/197

**Comment by melroy89** [comment]
> If the redirection is wrong, please update the website builder.

What is wrong with the website build to fix the broken links? And how to update the builder? I couldn't find the root-cause. It's only happening on the TypeScript doc page.

**Pull request merged**